backend/layers/persistence/persistence_mock.py

```diff
@@ -103,6 +104,11 @@ def _update_version_with_canonical(
                     datasets_to_include.append(dataset_version)
             # Replace 'datasets' array of Dataset version ids with 'datasets' array of actual Dataset versions
             copied_version.datasets = datasets_to_include
+            # Order by cell count if not custom ordered.
+            if not copied_version.datasets_custom_ordered:
+                copied_version.datasets.sort(
+                    key=lambda d: 0 if d is None or d.metadata is None else d.metadata.cell_count, reverse=True
+                )
```

Comment:
since we have this in at least two places now, let's consolidate into a common utils function, e.g. `sort_datasets_by_cell_count`